src/main/repositories/item_repository.py

```diff
@@ -0,0 +1,31 @@
+# This file has the item repository
+
+from src.main.repositories.generic_repository import GeneralRepository
+
+
+class ItemRepository(GeneralRepository):
+
+    def __init__(self):
+        self.__items = {}
+        self.__current_id = 0
+
+    def add(self, item):
+        item.id = self.__current_id
+        self.__items[item.id] = item
+        self.__current_id += 1
+        return item
```

Comment:
If you decided not to return the entity from within `update` then you should  follow the same  pattern within `add` to make your code consistent.